apps/frontend/src/modules/chats/slices/chats.slice.ts

```diff
@@ -74,34 +75,74 @@ const { actions, name, reducer } = createSlice({
 			state.dataStatus = DataStatus.REJECTED;
 		});
 
-		builder.addCase(addMessageToCurrentChat.fulfilled, (state, action) => {
-			state.currentChat = action.payload;
-		});
-		builder.addCase(updateChats.fulfilled, (state, action) => {
-			state.chats = action.payload;
+		builder.addCase(
+			addMessageToCurrentChat.fulfilled,
+			(state, { payload: { isMessageInCurrentChat, newMessage } }) => {
+				if (isMessageInCurrentChat && state.currentChat) {
+					state.currentChat = {
+						...state.currentChat,
+						messages: [newMessage, ...state.currentChat.messages],
+					};
+				}
+			},
+		);
+
+		builder.addCase(updateChats.fulfilled, (state, { payload: newMessage }) => {
+			state.chats = state.chats
+				.map((chat) => {
+					return chat.id === newMessage.chatId
+						? {
+								...chat,
+								lastMessage: newMessage,
+								unreadMessageCount:
+									chat.interlocutor.id === newMessage.senderUser.id
+										? Number(chat.unreadMessageCount) + NEW_MESSAGE_COUNT
+										: chat.unreadMessageCount,
+							}
+						: chat;
```

Comment:
what about writing like this?
```suggestion
					if(chat.id !== newMessage.chatId){
					return chat;
					}
					return {
								...chat,
								lastMessage: newMessage,
								unreadMessageCount:
									chat.interlocutor.id === newMessage.senderUser.id
										? Number(chat.unreadMessageCount) + NEW_MESSAGE_COUNT
										: chat.unreadMessageCount,
							};
```